src/test/run-make/thumb-none-qemu/Makefile

```diff
@@ -0,0 +1,42 @@
+-include ../../run-make-fulldeps/tools.mk
+
+# How to run this
+# $ ./x.py clean
+# $ ./x.py test --target thumbv7m-none-eabi src/test/run-make
+
+ifeq ($(TARGET),thumbv7m-none-eabi))
+
+# For cargo setting
+RUSTC := $(RUSTC_ORIGINAL)
+LD_LIBRARY_PATH := $(HOST_RPATH_DIR)
+# We need to be outside of 'src' dir in order to run cargo
+WORK_DIR := $(TMPDIR)
+HERE := $(shell pwd)
+
+
+# hint: we could set variables per $(TARGET) basis in order to support other targets.
+CRATE := lm3s6965evb
+CRATE_URL := https://github.com/japaric/lm3s6965evb
```

Comment:
I would prefer to use the [cortex-m-rt](https://github.com/rust-embedded/cortex-m-rt) crate here since it's maintained by the WG. You would have to:

- modify that crate's [memory.x](https://github.com/rust-embedded/cortex-m-rt/blob/05a4801e607e90edece279eaa32fd27e3bd76953/memory.x#L10-L11) file to match [this one](https://github.com/japaric/lm3s6965evb/blob/9eeea58826438e84d89e9691a1bb0f85b03d377c/memory.x#L4-L5)
- copy lm3s6965evb's main.rs into the examples directory of the cortex-m-rt crate

and if you add a .cargo/config file with [these lines](https://github.com/rust-embedded/cortex-m-quickstart/blob/a7f68f41cb1d9d4548f675b48873f73ee7434a37/.cargo/config#L1-L3) (but uncommented) to cortex-m-rt then `cargo run --example qemu --target $(TARGET)` should build the example and then run it on QEMU in a single step.

Feel free to send a PR and I'll review it!

Reply:
I'll do this first. (make a PR for `cortex-m-rt`)

Reply:
When moved to cortex-m-rt, semihosting did not worked for me.